### PR TITLE
Style fixes for Select-Xml tests

### PR DESCRIPTION
**Pull request by ThreeFive-O**
# PR Summary

This PR includes:
- Move Select-Xml tests from two different files into one.
- Update test descriptions.
- Update Pester syntax where applicable

## PR Context

Tests for a specific cmdlet should be in the corresponding test file.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Xml.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+			Remove-Item -Path 'env:xmltestfile'
+		}
+	}
+
+	It "Can throw for invalid XML file" {
+		$testfile = "TestDrive:\test.xml"
+		Set-Content -Path $testfile -Value "<a><b>"
+		$err = $null
+		Select-Xml -Path $testfile -XPath foo -ErrorVariable err -ErrorAction SilentlyContinue
+		$err.FullyQualifiedErrorId | Should -Be 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Xml.Tests.ps1]: The recommended pattern is to use `Should -Throw -ErrorId`. Have a look at: https://github.com/PowerShell/PowerShell/blob/a62eb62701091222a0adade43806e4fba97bae7b/docs/testing-guidelines/WritingPesterTests.md#writing-pester-tests

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by adityapatwardhan** [comment]
@ThreeFive-O Thanks for this PR! Could you update the tests to use the new recommended way of testing error?

**Comment by ThreeFive-O** [comment]
@adityapatwardhan I've found the following sentence in the Pester guideline under **"Pester Do and Don't"** you linked

> 7. Prefer Try-Catch for expected errors and check $_.fullyQualifiedErrorId (don't use should throw).

Is this still referring to a time, where Pester didn't support the syntax `Should -Throw -ErrorId`? If so, I guess we should update the guideline accordingly.

**Comment by adityapatwardhan** [comment]
Yes, that sentence should be removed. Please submit that change in a new PR if you can.

**Review by adityapatwardhan — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Xml.Tests.ps1 (hunk 1 above)
Comment (ThreeFive-O) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Xml.Tests.ps1]: Thank you for the great suggestion @adityapatwardhan

**Pull request merged**